ccdproc/bitmask.py

```diff
@@ -0,0 +1,473 @@
+        ignored by the function.
+
+        Examples
+        --------
+        The active flags can be given by their bit position (from right to
+        left)::
+
+            >>> from ccdproc import Bitmask
+            >>> bm = Bitmask(np.arange(8))
+            >>> bm.to_mask(bitpos=[1])
+            array([False, False,  True,  True, False, False,  True,  True], dtype=bool)
+
+        or by their bit value (``position ** 2``)::
+
+            >>> bm.to_mask(bitval=[2])
+            array([False, False,  True,  True, False, False,  True,  True], dtype=bool)
+
+        or if a ``name_map`` is given also by their name::
+
+            >>> bm.name_map['bad'] = 1
+            >>> bm.to_mask(bitname=['bad'])
+            array([False, False,  True,  True, False, False,  True,  True], dtype=bool)
+
+        It is also possible to specify only the **inactive** flags::
+
+            >>> bm.to_mask(bitname=['bad'], reverse_flag_interpretation=True)
+            array([False,  True, False,  True,  True,  True,  True,  True], dtype=bool)
+
+        and to specify multiple flags::
+
+            >>> bm.to_mask(bitpos=[0, 1])
+            array([False,  True,  True,  True, False,  True,  True,  True], dtype=bool)
+        """
+        bitmask = self.data
+        # Make sure one and only one of the three options to provide the
+        # "active" flags is actually given.
+        if sum(inp is not None for inp in (bitpos, bitval, bitname)) != 1:
+            raise TypeError('One and only one of "bitpos", "bitval" or '
+                            '"bitname" may be not "None".')
+
+        # So that the logic and validation is only applied once process
+        # first "bitname" to "bitpos" and then "bitpos" to "bitval". So each
+        # step is quite short and well defined and the logic can be implemented
+        # only in the "bitval" branch.
+
+        # Map the names to bit positions
+        if bitname is not None:
+            try:
+                bitpos = [self.name_map[name] for name in bitname]
+            except KeyError:
+                raise ValueError(
+                    'The "name(s)" {} were not found in "self.name_map" ({})'
+                    .format([name for name in bitname
+                             if name not in self.name_map],
+                            self.name_map))
+            bitname = None
+
+        # Map the bit positions to bit values
+        if bitpos is not None:
+            try:
+                bitval = [_POWERS[pos] for pos in bitpos]
+            except KeyError:
+                raise ValueError(
+                    'The "bit position(s)" {} are not in the allowed range of '
+                    'values.'.format([pos for pos in bitpos
+                                      if pos not in _POWERS]))
+            bitpos = None
+
+        # If bitname or bitvals were given these are now converted to bitvalues
+        # so we can do all the real processing in here.
+        if bitval is not None:
```

Comment:
This ``if`` is not necessary as ``bitval`` is always not ``None``. Shift the block below to the left.